ui/widget.js

```diff
@@ -390,6 +408,55 @@ $.Widget.prototype = {
 		return this._setOptions({ disabled: true });
 	},
 
+	_classes: function( element, keys, extra, add, object ) {
+		var full = [],
+			that = this;
+
+		object = object || this.options.classes;
+
+		function processClassString( classes, checkOption ) {
+			var current, i;
+			for ( i = 0; i < classes.length; i++ ) {
+				current = that.classObject[ classes[ i ] ] ? that.classObject[ classes[ i ] ] : $();
+				that.classObject[ classes[ i ] ] = current[ add ? "add" : "not" ]( element );
```

Comment:
We can avoid the ordering slowness with (untested):

```js
current = $( $.unique( current.get().concat( element.get() ) ) );
```

This would have decent parity with the required `.not()` handling too :-P